Backend/src/modules/live_data_processor/tools/cbl_parser/wow_vanilla/parser.rs

```diff
@@ -1051,7 +1051,7 @@ impl CombatLogParser for WoWVanillaParser {
                 participant.guild_args = Some((guild_name.to_string(), guild_rank_name.to_string(), guild_rank_index));
             }
 
-            if pet_name != "nil" {
+            if pet_name != "nil" && pet_name != "" {
```

Comment:
Use this function: https://doc.rust-lang.org/std/string/struct.String.html#method.is_empty-1